.github/scripts/format_dbt_test_failures.py

```diff
@@ -91,53 +95,58 @@ class FailedTestGroup:
     convenience methods for formatting those results for output to a report."""
 
     # Names of fields that are used for debugging
-    _debugging_field_names = [TEST_NAME_FIELD, DOCS_URL_FIELD]
-    # List that defines the order that diagnostic fields should appear in
-    # the output workbook
-    _diagnostic_field_order = [
-        SOURCE_TABLE_FIELD,
-        DESCRIPTION_FIELD,
-        TEST_NAME_FIELD,
-        DOCS_URL_FIELD,
+    _debugging_fieldnames = [TEST_NAME_FIELD, DOCS_URL_FIELD]
+    # Names of fields that identify the failing test
+    _test_metadata_fieldnames = [
+        *[SOURCE_TABLE_FIELD, DESCRIPTION_FIELD],
+        *_debugging_fieldnames,
+    ]
+    # Names of fields that are used for diagnostics
+    _diagnostic_fieldnames = [
         TAXYR_FIELD,
         PARID_FIELD,
         CARD_FIELD,
         TOWNSHIP_FIELD,
         WHO_FIELD,
         WEN_FIELD,
     ]
+    # The complete set of fixed fields
+    _fixed_fieldnames = [
+        *_test_metadata_fieldnames,
+        *_diagnostic_fieldnames,
+    ]
 
-    def __init__(self) -> None:
-        self._rows: typing.List[typing.Dict] = []
+    def __init__(
+        self, rows: typing.Optional[typing.List[typing.Dict]] = None
+    ) -> None:
+        self.raw_rows: typing.List[typing.Dict] = rows or []
```

Comment:
**question (dumb):** Is this just so we can use tests without any returned rows?

Reply:
By "this", do you mean the default initialization to an empty list? That's for convenience when extracting tests from Athena -- initializing with an empty list allows us to create an empty group and then use the same `update()` call to add the tests to the group whether or not the group already exists:

https://github.com/ccao-data/data-architecture/blob/1559fdc119826760c3d5a8e941ec5e3b220f6716/.github/scripts/format_dbt_test_failures.py#L491-L493

But we could rewrite this pretty easily if we wanted to require that rows exist:
```suggestion
        self, rows: typing.List[typing.Dict]
    ) -> None:
        self.raw_rows: typing.List[typing.Dict] = rows
```

```python
 if not failed_tests_by_category.get(category):
    failed_tests_by_category[category] = FailedTestGroup(failed_tests) 
else:
    failed_tests_by_category[category].update(failed_tests) 
```

Reply:
Nope fine as-is, I'm just a python dummy